File: Numbers/Calculator/C++/calculator.cpp
#include <iostream>
#include <string>   
#include <math.h>   

using namespace std;

// Калькулятор
// Calculator

std::string toString(int arg)
{
    return std::to_string(arg);
}

string calculator(int num1, int num2, char op) {

    string a = toString(num1) + " ";
    string b = toString(num2) + " = ";

    string symbols = "+-/*^";
    if (symbols.find(op) != std::string::npos) {
        switch (op) {
        case '+':
            return a + op + ' ' + b + toString(num1 + num2);
        case '-':
            return a + op + ' ' + b + toString(num1 - num2);
        case '*':
            return a + op + ' ' + b + toString(num1 * num2);
        case '/':
            return a + op + ' ' + b + toString(num1 / num2);
        case '^':
            return a + op + ' ' + b + toString(pow(num1, num2));

        }
    }
    else
        cout << "Please only type one of these characters: +, -, *, /!";


}


int main() {
    /*
    Main function
    - Get user's number
    - Calculate result
    - Print result
    */

    cout << "-- Calculator --" << endl;

    unsigned int num1,num2;
    char op;
    bool rep = true;
    string cont;

    do {
        cout << "\nType the first number: " << endl;
        cin >> num1;
        cout << "\nType the second number: " << endl;
        cin >> num2;
        cout << "What kind of operation would you like to do? \nChoose between + , -, *, / , ^ : " << endl;
        cin >> op;
        cout << calculator(num1, num2, op);
        cout << "\nContinue? (y,[n])" << endl;

        cin >> cont;
        if (cont != "y")
            rep = false;
    } while (rep);

    return 0;
}

File: Numbers/Calculator/Java/calculator.java
import java.util.Scanner;

public class calculator {

    public static void inputLimit()
    {
        // <input> -> nums and operator
        System.out.println("Type the first number: ");

        Scanner scanner = new Scanner(System.in);
        final int num1 = Integer.parseInt(scanner.nextLine());

        System.out.println("Type the second number: ");
        final int num2 = Integer.parseInt(scanner.nextLine());

        System.out.println("What kind of operation would you like to do? \nChoose between + , -, *, / , ^ : ");
        String op = scanner.nextLine();

        System.out.println(calc(num1, num2, op));

        System.out.println("\nEnter 'y' to continue: ");
        String answer = scanner.nextLine();

        if(answer.toLowerCase().equals("y"))
            inputLimit();
    }

    private static String calc(int num1, int num2, String op) {
        //  N -> fibonacci series
        String a = num1 + " ";
        String b = num2 + " = ";
        String symbols = "+-/*^";

        if(symbols.toLowerCase().contains(op.toLowerCase())){
            switch (op) {
                case "+":
                    return a + op + ' ' + b + (num1 + num2);
                case "-":
                    return a + op + ' ' + b + (num1 - num2);
                case "*":
                    return a + op + ' ' + b + (num1 * num2);
                case "/":
                    return a + op + ' ' + b + (num1 / num2);
                case "^":
                    return a + op + ' ' + b + (Math.pow(num1, num2));

            }
        }
        else
            return "Please only type one of these characters: +, -, *, /!";

        return "";
    }

    public static void main(String[] args)
    {
        // Main function
        // Get user's number
        // Calculate result
        // Print result

        System.out.println("-- Calculator -- ");

        inputLimit();
    }
}
